packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -89,8 +99,51 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Extract({path: directory}))
-      .on('close', () => {
+      .pipe(unzip.Parse())
+      .pipe(
+        new stream.Transform({
+          objectMode: true,
+          transform: async (entry, _, callback) => {
+            const fullPath = path.normalize(path.join(directory, entry.path))
+            if (!directory.endsWith(path.sep)) {
```

Comment:
```suggestion
            directory = path.resolve(directory)
            if (!directory.endsWith(path.sep)) {
```

Reply:
It is important for the comparison below. Otherwise it will fail with, for example, "/tmp/dest/file".startsWith("../dest/")

Reply:
We decided not to do the path.resolve above since this library could take relative paths!

Reply:
That was definitely failing when we switched to resolve temporarily.

Reply:
Ah, OK, `fullPath.startsWith(directory)` should work correctly then.